src/assets/wise5/services/tagService.ts

```diff
@@ -100,4 +102,10 @@ export class TagService {
     }
     return this.http.delete(`/api/projects/tag/${tag.text}`, { params: params }).subscribe();
   }
+
+  updateTag(tag: Tag): Subscription {
+    return this.http.put<Tag>(`/api/user/tag/${tag.id}`, tag).subscribe((tag) => {
+      this.tagChangedSource.next(tag);
+    });
+  }
```

Comment:
Do we need to return a Subscription? It looks like we don't do anything with it?
```suggestion
  updateTag(tag: Tag): void {
    this.http.put<Tag>(`/api/user/tag/${tag.id}`, tag).subscribe((tag) => {
      this.tagChangedSource.next(tag);
    });
  }
```

The other option is to return an Observable<Tag> and subscribe in the component.
```
  updateTag(tag: Tag): Observable<Tag> {
    return this.http.put<Tag>(`/api/user/tag/${tag.id}`, tag).pipe(
      tap((tag) => {
        this.tagChangedSource.next(tag);
      })
    );
  }
```